### PR TITLE
Bluetooth:  GATT: Allow discover all procedure

### PR DESCRIPTION
**Pull request by Vudentz**
This enables using bt_gatt_discover to discover all primary/secondary services.

### CODE REVIEW

File: subsys/bluetooth/shell/gatt.c [hunk 1]
```diff
@@ -159,10 +159,6 @@ int cmd_gatt_discover(int argc, char *argv[])
 	discover_params.end_handle = 0xffff;
 
 	if (argc < 2) {
```
Comment (joerchan) [inline subsys/bluetooth/shell/gatt.c]: Remove the goto done; completely while you are at it?
if ( argc > 1 ) { /* UUID handling */ }

Comment (Vudentz) [reply subsys/bluetooth/shell/gatt.c]: Done

**Review by jhedberg — APPROVED** [review]

**Comment by carlescufi** [comment]
CC @kapi-no @pdunaj @KAGA164

File: subsys/bluetooth/host/gatt.c [hunk 2]
```diff
@@ -1579,6 +1579,137 @@ static int gatt_read_type(struct bt_conn *conn,
...
+	const struct bt_att_read_group_rsp *rsp = pdu;
+	u16_t start_handle, end_handle = 0;
+	union {
+		struct bt_uuid uuid;
+		struct bt_uuid_16 u16;
+		struct bt_uuid_128 u128;
+	} u;
+
+	/* Data can be either in UUID16 or UUID128 */
+	switch (rsp->len) {
```
Comment (carlescufi) [inline subsys/bluetooth/host/gatt.c]: Doesn't `rsp->len` depend on the number of chars found?

File: subsys/bluetooth/host/gatt.c [hunk 3]
```diff
@@ -1579,6 +1579,137 @@ static int gatt_read_type(struct bt_conn *conn,
...
+		break;
+	case 20: /* UUID128 */
+		u.uuid.type = BT_UUID_TYPE_128;
+		break;
+	default:
+		BT_ERR("Invalid data len %u", rsp->len);
+		goto done;
+	}
+
+	/* Parse characteristics found */
```
Comment (kapi-no) [inline subsys/bluetooth/host/gatt.c]: Copy paste mistake :)
-> 
/* Parse services found */

**Review by kapi-no — APPROVED** [review]
@Vudentz Thank you for implementing this in response to my feature request #9713 ! I have just one minor comment

File: subsys/bluetooth/host/gatt.c (hunk 2 above)
Comment (carlescufi) [reply subsys/bluetooth/host/gatt.c]: Ah right, sorry. the name `len` confused me, this is actually the size of each Attribute Data pair included in the response.

**Review by carlescufi — APPROVED** [review]

File: subsys/bluetooth/host/gatt.c (hunk 3 above)
Comment (Vudentz) [reply subsys/bluetooth/host/gatt.c]: Right, will fix this.

**Pull request merged**